addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/comprehend/ComprehendServiceImpl.java

```diff
@@ -54,12 +65,16 @@ public DetectSentimentResult detectSentiment(String text, String languageCode) {
         }
 
         DetectSentimentRequest request = new DetectSentimentRequest().withText(text).withLanguageCode(languageCode);
-        DetectSentimentResult result = getClient().detectSentiment(request);
 
-        if (log.isDebugEnabled()) {
-            log.debug("DetectSentimentResult is " + result);
+        try {
+            DetectSentimentResult result = getClient().detectSentiment(request);
+            if (log.isDebugEnabled()) {
+                log.debug("DetectSentimentResult is " + result);
+            }
+            return result;
+        } finally {
+            awsMetrics.updateComprehendSentimentUnits(text.length() / 100L);
```

Comment:
Should we measure it, even if there was a network errors, or configuration error ?

Reply:
Indeed, I have rewrite all of them to be skipped if there is an exception